### ISSUE TITLE
Subtle timing bug in Event::update

### ISSUE THREAD

**Issue by dgroo** [JChristensen/Timer#33]
I came across a subtle timing bug in Event::update - took me a while to figure it out, though the fix is simple.  

tl:dr:  if you have alternating timer.after(...) calls made within a callback, they will often be called immediately on being scheduled.

Longer:
Basically what is happening is that "lastEventTime" ends up being after "now" in Event::update, which causes the calculation on the unsigned to roll over.

Changing the "if" in Event::update to something like this:

`if ((lastEventTime < now) && now - lastEventTime >= period)`

should fix the problem (preventing a case where the event happened in the future).

The attached .ino file should demonstrate the bug - with the fix, tick and tock alternate every five seconds.  Without it, tock() will always execute immediately after tick().

(If I get a chance I'll submit a pull request later, but the fix is so small thought I'd just post for now.)

Thanks for the lib!

[TimerBugDemo.zip](https://github.com/JChristensen/Timer/files/411432/TimerBugDemo.zip)
